### DISCUSSION TITLE
Debugging a single service orchestrated by Aspire

### DISCUSSION THREAD

**Discussion by KieranSunshine** [microsoft/aspire#4278] [Q&A]
Hey all, super pleased to see Aspire go to GA and I've been testing it out on some projects. So far it's been great!

I'm wondering what would be the easiest way to debug a single service that's normally orchestrated by Aspire? I can't see anything on the topic mentioned in the docs. I'm assuming it'd be something like running the service I need to debug externally (likely from an IDE) and somehow pointing the other Aspire projects to it. Honestly, this solution is probably fine. I just wasn't sure if there were any other 'better' approaches.

Cheers.

**Comment by timheuer** [comment]
Can you explain the setup here a bit more? You mentioned 'that's normally orchestrated' -- why wouldn't it be now then?

I'm assuming you are saying you'd still have something orchestrated w/Aspire (let's say ServiceA) that calls ServiceB, but that is NOT part of the Aspire environment?

If so, what is preventing ServiceB from being a part of it? Don't own the source, diff repo, etc? (that'd be good to know).  But given you say that you can still debug it I'm curious the source of ServiceB here?

**Reply by KieranSunshine** [reply]
So in our case we have a handful of services (REST API's) which depend on each other. Each of these services has it's own Git repository.

I've created an blank Aspire project and cloned each API into a `services` directory, the 'services' directory is ignored by Git. I haven't added project references from the Aspire project to the services, instead I'm just using `AddProject` and providing the path to the .csproj files. I have however added references from each project to the ServiceDefaults project for the time being.

After some adaptions to the services, running the Aspire project successfully brings up every service without a hitch, great stuff. 

Really what I'm trying to figure out is if I want to debug Service A which depends on Service B and Service C. Can I run Service B and Service C through my Aspire project, whilst running/debugging Service A from my IDE?

**Reply by davidfowl** [reply]
You’d have to attached after the fact using the IDE. It sounds like you want an easier way to attach or at least say which service to attach to on run
